### PR TITLE
r3.in.v5d, r3.out.v5d: Deprecate CRAY HW specific code

### PR DESCRIPTION
**Pull request by ymdatta**
Currently, there is no sign either from github issues or grass dev archive that many people are using CRAY code which was introduced 25 years ago. As part of fixing a bug, we found that it was hard to find reference materials related to CRAY, due to it being old and not in usage anymore. For this reason, deprecate CRAY code and corresponding complexity that comes with it.

### CODE REVIEW

File: raster3d/r3.in.v5d/v5d.h [hunk 1]
```diff
@@ -43,8 +43,8 @@
  * Define our own 1 and 2-byte data types.  We use these names to avoid
  * collisions with types defined by the OS include files.
  */
-typedef unsigned char V5Dubyte;   /* Must be 1 byte, except for cray */
-typedef unsigned short V5Dushort; /* Must be 2 byte, except for cray */
+typedef unsigned char V5Dubyte;   /* Must be 1 byte */
+typedef unsigned short V5Dushort; /* Must be 2 byte */
```
Comment (nilason) [inline raster3d/r3.in.v5d/v5d.h:47]: I wonder if it wouldn't be better to use C99's `int8_t` and `int16_t` (from stdint.h) here? That should by definition be the same on any platform.

Comment (wenzeslaus) [reply raster3d/r3.in.v5d/v5d.h:47]: I agree. Please, open a separate issue or PR for that. I intend to merge this just with the removal.

**Review by wenzeslaus — CHANGES_REQUESTED** [review]
This was more difficult to review then I expected from a code removal change. I think there is one issue, but otherwise it's good.

File: raster3d/r3.out.v5d/binio.c [hunk 2]
```diff
@@ -420,15 +202,13 @@ int read_float4(int f, float *x)
...
     }
-#else
+#endif
     if (read(f, x, 4) == 4) {
         return 1;
     }
     else {
         return 0;
     }
-#endif
```
[the remark this answers is not in the document]
Comment (ymdatta) [reply raster3d/r3.out.v5d/binio.c:430]: Initially, we had something like this:

```
if CRAY {
} else {
  if LITTLE {
  } else {
  }
}
```

after removing CRAY conditional code, it became:

```
if LITTLE {
A;
B;
} else {
C;
D;
}
```

Now, when the execution enters the ```LITTLE``` conditional block, it always returns out from the function. So, technically, it's the same as:

```
if LITTLE {
A;
B;
}

C;
D;
```

as we only execute C and D, if it's not `LITTLE`, which is same as having it under else.

I just thought that this is more simpler, if it's not I can change it. Let me know what do you think about it.

[the remark this answers is not in the document]
Comment (wenzeslaus) [reply raster3d/r3.out.v5d/binio.c:430]: This makes sense. That's what we would do for if-statements, so we can use it for the defines as well.

**Review by wenzeslaus — APPROVED** [review]

**Pull request merged**